src/frontends/sql/decoder_context_test.cc

```diff
@@ -127,4 +135,95 @@ TEST(DecoderContextDeathTest, DecoderContextValueRegistryDeathTest) {
                "Could not find a value with id 3.");
 }
 
+class DecodeMergeOpTest
+    : public TestWithParam<std::tuple<std::vector<Value>, std::vector<Value>>> {
+};
+
+static bool BeginsWith(absl::string_view str1, absl::string_view str2) {
+  if (str2.size() > str1.size()) return false;
+  return str1.substr(0, str2.size()) == str2;
+}
+
+static std::optional<uint64_t> ExtractIdxAfterPrefix(absl::string_view str,
+                                                     absl::string_view prefix) {
+  if (!BeginsWith(str, prefix)) return std::nullopt;
+  uint64_t result = 0;
+  std::string index_str(str.substr(prefix.size()));
+  EXPECT_NO_THROW({ result = std::stoul(index_str); });
+  return result;
+}
+
+TEST_P(DecodeMergeOpTest, DecodeMergeOpTest) {
```

Comment:
Ideally, unit tests should not have this much logic as it makes debugging failures very difficult. Can we use any of the gmock matchers?

Reply:
This has been slimmed down considerably, as most of the logic has moved to `testing::MergeOperationView`.